### PR TITLE
function for calculating image polygon vertices

### PR DESCRIPTION
**Pull request by PKukic**

### CODE REVIEW

File: void/tests/data/test_math.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import os
+import unittest
+from unittest import mock
+
+import docopt
+
+from void import math_utils
+
+class CalculatePolyTest(unittest.TestCase):
```
Comment (frnhr) [inline void/tests/data/test_math.py]: duplicate file, right?

File: void/tests/test_reducer.py [hunk 2]
```diff
@@ -76,10 +76,10 @@ def test_main(self, p_sys, *_):
         p_sys.stdin = ['void/tests/data/test_unflagged.fit']
         reducer.main()
         expected_output = (
-            '{"date_obs": "2019-01-09T04:47:09.360", "exposition": 60.0, '
-            '"focus": 4408, "ra_center": "11 11 13.00", '
-            '"dec_center": "+63 18 32.0", "x_deg_size": 0.7300444444444444, '
-            '"y_deg_size": 0.7300444444444444}\n'
+            '{"date_obs": "2019-01-09T04:47:09.360", '
+            '"exposition": 60.0, "focus": 4408, "ra_center": 167.81972847, '
```
Comment (frnhr) [inline void/tests/test_reducer.py]: The correct term is "exposure", not "exposition". 🙏 Search & replace?

File: void/math_utils.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+"""
+Miscellaneous math functions
+"""
+
+import numpy as np
+
+
+def calculate_poly(image_center, image_x, image_y, pos_angle):
+    """ Calculate image border points. """
```
Comment (frnhr) [inline void/math_utils.py]: "border points" should be "vertices" (singular: vertex) or "corners"

File: void/tests/test_math.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+from void import math_utils
+
+
+class CalculatePolyTest(unittest.TestCase):
+    def test_calculate_poly(self):
+        values = math_utils.calculate_poly((0, 0), 3, 3, 315)
+        expected = [[-1.29893408e-16, -2.12132034e+00],
+                    [-2.12132034e+00, 0.00000000e+00],
+                    [1.29893408e-16, 2.12132034e+00],
+                    [2.12132034e+00, 0.00000000e+00]]
```
Comment (frnhr) [inline void/tests/test_math.py]: We could try to round those numbers so that we are not dealing with `e-16` which is really a `0`.
Maybe  calling `round(x, 4)` on each of those values:
```python
values = math_utils.calculate_poly((0, 0), 3, 3, 315)
values = [list(map(lambda x: round(x, 4), x)) for x in values]  # just rounding to 3 decimals
expected = [[0, -2.1213],
            [-2.1213, 0],
            [0, 2.1213],
            [2.1213, 0]]
```

File: void/tests/test_math.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import unittest
+
+from void import math_utils
+
+
+class CalculatePolyTest(unittest.TestCase):
+    def test_calculate_poly(self):
```
Comment (frnhr) [inline void/tests/test_math.py]: Let's add a few more test cases just to be sure that edge cases work. We could do something like
```python
values = math_utils.calculate_poly((0, 0), 3, 3, 0)
values = [list(map(lambda x: round(x, 4), x)) for x in values]
expected = [[1.5, -1.5],
            [-1.5, 1.5],
            [1.5, 1.5],
            [-1.5, -1.5]]
``` 

```python
values = math_utils.calculate_poly((0, 0), 3, 3, 90)
values = [list(map(lambda x: round(x, 4), x)) for x in values]
expected = [[1.5, -1.5],
            [-1.5, 1.5],
            [1.5, 1.5],
            [-1.5, -1.5]]
``` 

And one that is not a square:
```python
values = math_utils.calculate_poly((0, 0), 4, 2, 90)
values = [list(map(lambda x: round(x, 4), x)) for x in values]
expected = [[1, -2],
            [-1, 2],
            [1, 2],
            [-1, -2]]
```

Comment (frnhr) [reply void/tests/test_math.py]: And then to avoid repeating that rounding:

```python
@staticmethod
def round(values):
     return [list(map(lambda x: round(x, 4), x)) for x in values]

def test_something....
    values = math_utils.calc...
    values = self.round(values)
    expected = ...

File: void/reducer.py [hunk 6]
```diff
@@ -39,30 +39,37 @@ def read_header_data(fits_fname):
...
 
-        x_deg_size = float(x_pix_size * x_binning * x_scale) / 3600
-        y_deg_size = float(y_pix_size * y_binning * y_scale) / 3600
+        x_deg_size = float(x_pix_size * x_scale)
+        y_deg_size = float(y_pix_size * y_scale)
+
+        pos_angle = header_dict['PA']
+
+        # Limiting magnitude
+        # mag_norm = header_dict['starZMAG']
```
Comment (frnhr) [inline void/reducer.py]: card name is `ZMAG`, not `starZMAG`, check this:
```
In [1]: from astropy.io import fits                                                                                                                                                                                                                                                                                                                                  

In [2]: f = fits.open('void/tests/data/test_unflagged.fit')                                                                                                                                                                                                                                                                                                          

In [3]: h = f[0].header                                                                                                                                                                                                                                                                                                                                              

In [4]: h['ZMAG']                                                                                                                                                                                                                                                                                                                                                    
Out[4]: 19.577995437

In [5]: h                                                                                                                                                                                                                                                                                                                                                            
Out[5]: 
SIMPLE  =                    T                                                  
BITPIX  =                   16 /8 unsigned int, 16 & 32 int, -32 & -64 real     
NAXIS   =                    2 /number of axes                                  
...                 
RA      = '11 11 13.00'        / [hms J2000] Target right ascension             
DEC     = '+63 18 32.0'        / [dms +N J2000] Target declination              
FWHM    =   2.50253389478E+000 / [pixels] Mean Full-Width-Half-Max of image star
ZMAG    =   1.95779954370E+001 / Mag zero point for 1 sec exposure              
...
```

Do you see the ⭐️? :trollface:

Comment (PKukic) [reply void/reducer.py]: nice, got it

File: void/tests/test_math.py (hunk 5 above)
Comment (PKukic) [reply void/tests/test_math.py]: ok

File: void/tests/test_math.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    def test_calculate_45(self):
+        values = math_utils.calculate_poly((0, 0), 3, 3, 315)
+        values = self._round(values)
+        expected = [[0, -2.1213], [-2.1213, 0], [0, 2.1213], [2.1213, 0]]
+        expected = np.asarray(expected)
+        expected = math_utils.sort_ndarray(expected)
+        try:
+            np.testing.assert_almost_equal(values, expected, decimal=4)
+            res = True
+        except AssertionError as err:
```
Comment (frnhr) [inline void/tests/test_math.py]: Way to go, `np.testing` is nice 👌!
But why are we catching `AssertationError` only to raise it from a different function? Can we replace the whole `try...except...` with just `np.testing.assert_almost...`?

Comment (PKukic) [reply void/tests/test_math.py]: Done.
